sigal/writer.py

```diff
@@ -112,8 +113,10 @@ def generate_context(self, album):
 
     def write(self, album):
         """Generate the HTML page and save it."""
-
-        page = self.template.render(**self.generate_context(album))
+        context = self.generate_context(album)
+        for receiver in signals.before_render.receivers_for(context):
+            context = receiver(context)
```

Comment:
Any reason why you need to iterate on receivers instead on sending context directly ?
```suggestion
        signals.before_render.send(context)
```

Reply:
Right... I was looking at here

https://github.com/saimn/sigal/blob/0d8a3c8779ab32855575bbdafba02645d4b0e290/sigal/image.py#L137-L138

thinking that likewise the `context` dict can be modified by plugins so it should be returned and reassigned... without realizing that a dict is mutable and is passed by reference.

Reply:
Fixed in 2c4fdd6.